pkg/controller/egress/controller.go

```diff
@@ -402,6 +402,7 @@ func (c *EgressController) enqueueEgressGroup(key string) {
 // addEgress processes Egress ADD events and creates corresponding EgressGroup.
 func (c *EgressController) addEgress(obj interface{}) {
 	egress := obj.(*egressv1alpha2.Egress)
+	c.updateEgressStatusCondition(obj, egressv1alpha2.IPAllocated, false)
```

Comment:
Event handlers are not the appropriate place to mak external call. It needs to be fast and not blocking as it's called sequently for all watchers.
And it shouldn't always update the egress once with a false condition. Instead, process the Egress first, update the condition to True when it gets an IP and false when it doesn't. 
`syncEgress` is a better place.